### PR TITLE
Create spree payment sources

### PR DESCRIPTION
**Pull request by GrzegorzStochel**

### CODE REVIEW

File: core/db/migrate/20220113052823_create_payment_sources.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class CreatePaymentSources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_payment_sources do |t|
+      t.string :gateway_source_id
```
Comment (GrzegorzStochel) [inline core/db/migrate/20220113052823_create_payment_sources.rb:4]: in spree_credit_cards we save that value in `gateway_payment_profile_id` . Should I use the same name here?

Comment (damianlegawiec) [reply core/db/migrate/20220113052823_create_payment_sources.rb:4]: @GrzegorzStochel let's use the same, it's going to be easier to remember this

File: core/db/migrate/20220113052823_create_payment_sources.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+      if t.respond_to? :jsonb
+        t.jsonb :public_metadata
+        t.jsonb :private_metadata
+      else
+        t.json :public_metadata
+        t.json :private_metadata
+      end
+
+      t.timestamps
```
Comment (damianlegawiec) [inline core/db/migrate/20220113052823_create_payment_sources.rb:18]: @GrzegorzStochel let's add indexes for 

`type` &  `gateway_source_id` - shouldn't this be unique?

Comment (GrzegorzStochel) [reply core/db/migrate/20220113052823_create_payment_sources.rb:18]: we didn't have uniq on CC, but theoretically, it won't hurt to have that not to double the same records

File: core/db/migrate/20220113052823_create_payment_sources.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class CreatePaymentSources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_payment_sources do |t|
+      t.string :gateway_payment_profile_id, index: { unique: true }
+      t.string :type, index: true
+
+      t.references :payment_method, index: true, foreign_key: { to_table: :spree_payment_methods }
+      t.references :user, index: true, foreign_key: { to_table: :spree_users }
```
Comment (damianlegawiec) [inline core/db/migrate/20220113052823_create_payment_sources.rb:8]: @GrzegorzStochel I think we should skip the foreign key here as the user is optional, right?

Comment (GrzegorzStochel) [reply core/db/migrate/20220113052823_create_payment_sources.rb:8]: I think the purpose of the foreign key is to raise the error while we try to delete parents if there are children records in the DB. Does `optional` change anything here?

Comment (damianlegawiec) [reply core/db/migrate/20220113052823_create_payment_sources.rb:8]: You are correct, my bad!

File: core/db/migrate/20220113052823_create_payment_sources.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class CreatePaymentSources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_payment_sources do |t|
+      t.string :gateway_payment_profile_id, index: { unique: true }
```
Comment (damianlegawiec) [inline core/db/migrate/20220113052823_create_payment_sources.rb:4]: @GrzegorzStochel this unique index should be in conjunction with type, there is a slight chance that the ID could be the same for 2 different gateways

Comment (GrzegorzStochel) [reply core/db/migrate/20220113052823_create_payment_sources.rb]: done

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**